api/src/identity-access-management/application/http-error-mapper-configuration.js

```diff
@@ -20,6 +21,10 @@ const authenticationDomainErrorMappingConfiguration = [
     name: DifferentExternalIdentifierError.name,
     httpErrorFn: (error) => new HttpErrors.ConflictError(error.message),
   },
+  {
+    name: InvalidOrAlreadyUsedEmailError.name,
+    httpErrorFn: (error) => new HttpErrors.UnprocessableEntityError(error.message, error.code),
```

Comment:
```suggestion
    httpErrorFn: (error) => new HttpErrors.BadRequestError(error.message, error.code),
```

En effet on peut lire que le code statut `422` qui est utilisé pour la `HttpErrors.UnprocessableEntityError` est à utiliser dans des situations qui n'ont aucun rapport avec ce dont il est question ici. Plus généralement ce mapping est complètement erroné dans Pix en général. J'ai prévu de m'en occuper un jour en Tech Times.

cf. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses
cf. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
cf. https://fr.wikipedia.org/wiki/Liste_des_codes_HTTP

Aussi un `BadRequestError` qui produit un `400` est mieux adaptée.

Reply:
Ce point sera traité dans un ticket d'amélioration continue :-)